src/CommunityToolkit/Xamarin.CommunityToolkit/Behaviors/EventToCommandBehaviorGeneric.shared.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+
+namespace Xamarin.CommunityToolkit.Behaviors
+{
+	/// <summary>
+	/// This <see cref="EventToCommandBehavior"/> cast the sender object to a specific type defined by the user.
+	/// </summary>
+	/// <typeparam name="TType">The type that you want to receive in your <see cref="Xamarin.Forms.Command"/> </typeparam>
+	public sealed class EventToCommandBehavior<TType> : EventToCommandBehavior
+	{
+		protected override void OnTriggerHandled(object? sender = null, object? eventArgs = null)
+		{
+			var parameter = CommandParameter
+				?? EventArgsConverter?.Convert(eventArgs, typeof(object), null, null)
+				?? eventArgs;
+
+			if (parameter?.GetType() != typeof(TType))
```

Comment:
Let's use C#9.0 Pattern Matching here to improve readability:

```csharp
if (parameter is not TType)
```

Reply:
Nice catch!

Reply:
@brminnick as I see here @pictos checks that type is exactly the same as TType.

**is** operator will also return **true** if the parameter type is a subtype of TType.

Reply:
Good point.

@pictos Do you want to allow or prevent inherited types?

If we do want to allow inherited types, we should add a Unit Test for it. Something like this:

```csharp
[Test]
public void NoExceptionWhenTheEventArgsAreNotNull_InheritedType()
{
	var vm = new ViewModelCoffe();
	var behavior = new EventToCommandBehavior<Coffee>
	{
		EventName = nameof(ListView.ItemTapped),
		EventArgsConverter = new ItemSelectedEventArgsConverter(),
		Command = vm.SelectedCommand
	};

	Assert.Null(vm.CoffeeName);
	var coffe = new Latte { Id = 1, Name = "Latte" };
	var eventArgs = new SelectedItemChangedEventArgs(coffe, 1);

	var notNullArgs = new object?[] { null, eventArgs };

	TriggerEventToCommandBehavior(behavior, notNullArgs);

	Assert.AreEqual(coffe.Name, vm.CoffeeName);
}

class Latte : Coffee
{
}
```